dev-tools/mage/dockerbuilder.go

```diff
@@ -173,22 +180,42 @@ func (b *dockerBuilder) expandDockerfile(templatesDir string, data map[string]in
 	return nil
 }
 
-func (b *dockerBuilder) dockerBuild() (string, error) {
-	tag := fmt.Sprintf("%s:%s", b.imageName, b.Version)
+// dockerBuild runs "docker build -t t1 -t t2 ... buildDir"
+// returns the main tag and the more tags
+// returns an error if the command fails
+func (b *dockerBuilder) dockerBuild() (string, []string, error) {
+	mainTag := fmt.Sprintf("%s:%s", b.imageName, b.Version)
 	// For Independent Agent releases, replace the "+" with a "." since the "+" character
 	// currently isn't allowed in a tag in Docker
 	// E.g., 8.13.0+build202402191057 -> 8.13.0.build202402191057
-	tag = strings.Replace(tag, "+", ".", 1)
+	mainTag = strings.Replace(mainTag, "+", ".", 1)
 	if b.Snapshot {
-		tag = tag + "-SNAPSHOT"
+		mainTag = mainTag + "-SNAPSHOT"
 	}
 	if repository := b.ExtraVars["repository"]; repository != "" {
-		tag = fmt.Sprintf("%s/%s", repository, tag)
+		mainTag = fmt.Sprintf("%s/%s", repository, mainTag)
+	}
+
+	moreTags := []string{}
+	if commit, found := b.ExtraVars["commit"]; found && b.DockerVariant == Service {
+		if len(commit) >= 12 {
+			moreTags = append(moreTags, fmt.Sprintf("%s:%s", b.imageName, "git-"+commit[:12]))
+		}
+	}
```

Comment:
ExtraVars contains raw values, expanded values are inside evalContext, use that one and replace also one for repository above 

```suggestion
	moreTags := []string{}
	if commitI, found := b.evalContext["commit"]; found && b.DockerVariant == Service {
		commit, ok := commitI.(string)
		if ok && len(commit) >= 12 {
			moreTags = append(moreTags, fmt.Sprintf("%s:git-%s", b.imageName, commit[:12]))
		}
	}
```